### PR TITLE
Performance improvement when adding multiple accounts

### PR DESCRIPTION
**Pull request by blurpesec**
### Description

Batch create/update contacts to decrease the number of updates we make when adding multiple accounts.

### CODE REVIEW

File: src/services/Store/Contact/useContacts.tsx [hunk 1]
```diff
@@ -65,10 +67,24 @@ function useContacts() {
...
 
+  const createContacts = (items: Contact[]) => {
+    dispatch(
+      createManyContacts(
+        items.map((item) => 
+          ({ ...item, uuid: generateDeterministicAddressUUID(item.network, item.address) })
+        )
+      )
+    );
+  };
```
Comment (vilsbole) [inline src/services/Store/Contact/useContacts.tsx:78]: Caller should be obnoxious of the uuid format.
Use action creator for setting the uuid.

```
type createContact(items: Contact[], generateUUID: (any) => TUuid = generateDeterministicUUID)
```

File: src/services/Store/Contact/useContacts.tsx [hunk 2]
```diff
@@ -118,7 +134,9 @@ function useContacts() {
     contacts,
     contactRestore,
     createContact,
+    createContacts,
```
Comment (vilsbole) [inline src/services/Store/Contact/useContacts.tsx:137]: We are working to remove the hooks and let the components connect directly to the store.

File: src/services/Store/StoreProvider.tsx [hunk 3]
```diff
@@ -120,7 +121,12 @@ export const StoreProvider: React.FC = ({ children }) => {
   const { assets } = useAssets();
   const { settings, updateSettingsAccounts } = useSettings();
   const { networks } = useNetworks();
-  const { createContact, contacts, getContactByAddressAndNetworkId, updateContact } = useContacts();
+  const {
+    createContacts,
```
Comment (vilsbole) [inline src/services/Store/StoreProvider.tsx:125]: Prefer `connect` over hook.

File: src/services/Store/StoreProvider.tsx [hunk 4]
```diff
@@ -389,24 +395,41 @@ export const StoreProvider: React.FC = ({ children }) => {
...
+                  network: rawAccount.networkId,
+                  uuid: generateUUID()
+                }
+              ]
+            };
+          }
+          return acc;
+        },
+        { newContacts: [] as ExtendedContact[], updatedContacts: [] as ExtendedContact[] }
+      );
```
Comment (vilsbole) [inline src/services/Store/StoreProvider.tsx:430]: This looks like a create or update logic.
It should be the responsibility of the slice.
Move to redux.

FYI: there appears to already be a createOrUpdate defined

**Review by vilsbole — CHANGES_REQUESTED** [review]

**Comment by FrederikBolding** [comment]
Closing in favor of #4009

**Pull request closed without merging**